src/sentry/auth/access.py

```diff
@@ -47,68 +54,141 @@ def get_permissions_for_user(user_id: int) -> FrozenSet[str]:
     return cast(FrozenSet[str], union)
 
 
-@dataclass(frozen=True, eq=True)
-class SsoState:
-    is_required: bool
-    is_valid: bool
+class Access(abc.ABC):
+    @property
+    @abc.abstractmethod
+    def sso_is_valid(self) -> bool:
+        pass
```

Comment:
Fair enough. 🙂

Reply:
Thinking about it, it might be possible to write a python autoformatter using the ast module that rewrites source files for this.  A method of a class decorated by abstractmethod or abc.abstractmethod automatically receives such a body.  Could be interesting.